qiita_db/metadata_template/prep_template.py

```diff
@@ -111,7 +108,7 @@ def create(cls, md_template, raw_data, study, data_type,
 
         # Get a connection handler
         conn_handler = SQLConnectionHandler()
-        queue_name = "CREATE_PREP_TEMPLATE_%d" % raw_data.id
+        queue_name = "CREATE_PREP_TEMPLATE_%d" % study.id
```

Comment:
These are just the queues though. Given that tornado is single threaded, is it possible for this block of code to be executed twice without the first queue draining?